commit 91f07cdccd8156b58bf33379e29bda6f39df4ba1
Author: Robert Mustacchi <rm@joyent.com>
Date:   2019-08-30T14:16:17+00:00 (5 weeks ago)
    
    OS-7976 debug build needs some NULL pointer fixes
    Reviewed by: John Levon <john.levon@joyent.com>
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
    Approved by: John Levon <john.levon@joyent.com>

diff --git a/usr/src/uts/common/brand/lx/syscall/lx_aio.c b/usr/src/uts/common/brand/lx/syscall/lx_aio.c
index 1b673066e6..6748245db8 100644
--- a/usr/src/uts/common/brand/lx/syscall/lx_aio.c
+++ b/usr/src/uts/common/brand/lx/syscall/lx_aio.c
@@ -248,7 +248,7 @@ lx_io_cp_hold(lx_aio_context_t cid)
 
 	if (lxpd->l_io_ctxs == NULL) {
 		ASSERT(lxpd->l_io_ctx_cnt == 0);
-		ASSERT(lxpd->l_io_ctxpage == NULL);
+		ASSERT(lxpd->l_io_ctxpage == (uintptr_t)NULL);
 		goto bad;
 	}
 
@@ -622,7 +622,7 @@ lx_io_setup(uint_t nr_events, void *ctxp)
 		uintptr_t ctxpage;
 
 		ASSERT(lxpd->l_io_ctx_cnt == 0);
-		ASSERT(lxpd->l_io_ctxpage == NULL);
+		ASSERT(lxpd->l_io_ctxpage == (uintptr_t)NULL);
 
 		ttolwp(curthread)->lwp_errno = 0;
 		ctxpage = (uintptr_t)smmap64(0, PAGESIZE, PROT_READ,
